src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/process/SynchronousOutputProcessRunner.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.cloudsdk.internal.process;
+
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessExitListener;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessOutputLineListener;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessStartListener;
+import com.google.common.collect.Lists;
+
+/**
+ * A ProcessRunner that runs synchronous processes and makes their stderr and stdout streams
+ * available as Strings.
+ */
+public class SynchronousOutputProcessRunner extends DefaultProcessRunner implements ProcessRunner {
```

Comment:
I'm still convinced this class is redundant. It also manages to completely ignore the listeners set by the user when sdk is constructed.